### ISSUE TITLE
Cypress does not pick up Node version set by Action

### ISSUE THREAD

**Issue by sergiubcn** [cypress-io/github-action#467]
**Issue:** Cypress does not pick up the NodeJS version set by another GH Action

**Versions:** Cypress 9.x

**Steps to reproduce:**
1. Configure a GH Actions pipeline as follows: (a) set up a NodeJS version other than `12` (which is used by default in the Cypress GH Action) and (b) execute any Cypress test.
2. Notice the NodeJS version that Cypress uses.

**Prerequisites:** (a) a GH repo and (b) a GH workflow `yml` config.

**Conclusions:** Regardless of the NodeJS version set by `actions/setup-node@v2` Cypress still uses the default `12`. This is working as expected with Cypress 8.x.

**Example:** https://github.com/sergiubcn/cy-ci-node/runs/4467813743?check_suite_focus=true
https://github.com/sergiubcn/cy-ci-node/blob/master/.github/workflows/cicd.yml

**Note:** This does not reproduce locally if setting various NodeJS versions via `nvm use`. 
If it actually turns out to be an issue with Cypress I'll gladly move this ticket to that repo.

**Comment by magnetikonline** [comment]
Thanks for reporting @sergiubcn - just like to add we're seeing exactly the same behaviour in our workflows.

**Comment by BioCarmen** [comment]
Is that fixed now. I am seeing the same issue when I update cypress to 9.*

**Comment by unickq** [comment]
Cypress 9.2.0

```yaml
jobs:
  cypress-run:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Setup Node
        uses: actions/setup-node@v1
        with:
          node-version: 16
      - run: node -v

      - name: Cypress run
        uses: cypress-io/github-action@v2
```

![image](https://user-images.githubusercontent.com/6696821/148548062-89119402-bf38-41bd-ae95-fb52046e8e69.png)

**Comment by rbayliss** [comment]
There's another issue filed for this, with a proposed (temporary) workaround: https://github.com/cypress-io/github-action/issues/489#issuecomment-1021379037

**Comment by MikeMcC399** [comment]
@sergiubcn 

This is an old issue and I wanted to give you some feedback that what you describe is the correct way that github-action works. It is the consequence of using [Cypress Module API](https://docs.cypress.io/guides/guides/module-api) introduced in `github-action@v2`. Calling Cypress this way allows Cypress to return detailed results to `github-action`. It does however tie Cypress to using the version of Node.js which is defined in [action.yml](https://github.com/cypress-io/github-action/blob/38817732586351d080f5ea0a828482f78c4cd370/action.yml#L99-L101). For `github-action@v5` this is currently Node.js `16.16.0`. GitHub currently only offers the `node16` environment for the [runs.using](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-javascript-actions) option of GitHub JavaScript Actions.

It is now documented in the [README > Node.js](https://github.com/cypress-io/github-action/blob/master/README.md#nodejs) section.

I suggest to now close this issue as as far as I can see there is nothing which should be corrected in `github-action` regarding this topic.

**Comment by sergiubcn** [comment]
Thanks for the explanation @MikeMcC399 !
It is a bit counterintuitive, but it definitely helps that it's documented.
I'm closing this issue because it's not blocking me in any way, but if other participants feel this is unsatisfactory for their situation they can reopen it.

**Comment by tarektarho** [comment]
Hello, 

This issue start to happen again since, last Friday 5th of July. 

<img width="922" alt="Screenshot 2024-07-08 at 14 39 21" src="https://github.com/cypress-io/github-action/assets/18512695/d67c2eb4-257e-4a01-a012-e46a8854dfc3">

**Comment by MikeMcC399** [comment]
@tarektarho 

Which version of webpack are you using?

If you still need assistance, please open a new issue and post your full workflow and your `package.json`.
If you have logs, please post them as text, not screenshots.

**Comment by ungarida** [comment]
This is still a problem. We are using `setup-node` with v20.10 we print it and it confirms it is used. But then Cypress prints that is using v.20.13
![image](https://github.com/user-attachments/assets/c8721e46-0329-4efe-a94e-78f92c4d1770)

**Comment by MikeMcC399** [comment]
@ungarida 

What you are seeing is correct. It is described in https://github.com/cypress-io/github-action/blob/master/README.md#usage

The Node.js version reported by Cypress comes from the GitHub runner. If you are seeing `v20.13.1` it means that under your logs in the section "Set up job" you will probably see:
> Current runner version: '2.320.0'

which is defined by `NODE20_VERSION="20.13.1"`  in https://github.com/actions/runner/blob/v2.320.0/src/Misc/externals.sh
